src/NHibernate/Cfg/Configuration.cs

```diff
@@ -2479,6 +2482,25 @@ private IEnumerable<IPersistentIdentifierGenerator> IterateGenerators(Dialect.Di
 			return generators.Values;
 		}
 
+		/// <summary>
+		/// Returns a quoted name(schema/catalog) if needed
+		/// </summary>
+		/// <param name="name">The schema name</param>
+		/// <param name="dialect">The instance of dicalect to use</param>
+		/// <returns></returns>
+		private string GetQuotedName(string name, Dialect.Dialect dialect)
```

Comment:
if the catalog/schema is not quoted and a reserved keyword is used it will fail. Don't you think?